File: bounded-contexts/dotnet/MyShop.Services/ShoppingCart/CartItem.cs
using MyShop.Entities;
using System;
using System.Collections.Generic;
using System.Text;

namespace MyShop.Services.ShoppingCart
{
    public class CartItem
    {
        public CartItem(ItemId itemId, int quantity)
        {
            ItemId = itemId;
            Quantity = quantity;
        }

        public ItemId ItemId { get; }
        public int Quantity { get; }
    }
}

File: bounded-contexts/java/src/main/java/p1/myshop/services/shoppingcart/CartItem.java
package p1.myshop.services.shoppingcart;

import p1.myshop.entities.ItemId;
import p1.myshop.entities.Quantity;

public class CartItem {
    public final ItemId itemId;
    public final Quantity quantity;

    public CartItem(ItemId itemId, Quantity quantity) {
        this.itemId = itemId;
        this.quantity = quantity;
    }

    public ItemId itemId() {
        return itemId;
    }
}
